R/GitStats.R

```diff
@@ -0,0 +1,505 @@
+#' @importFrom R6 R6Class
+#' @importFrom data.table rbindlist :=
+#' @importFrom purrr map
+#' @importFrom tibble tibble
+#'
+#' @title A statistics platform for Git clients
+#' @description An R6 class object with methods to derive information from multiple Git platforms.
+#'
+GitStats <- R6::R6Class("GitStats",
+  public = list(
+
+    #' @field clients A list of API connections information.
+    clients = list(),
+
+    #' @description Create a new `GitStats` object
+    #' @return A new `GitStats` object
+    initialize = function() {
+
+    },
+
+    #' @field team A character vector containing team members.
+    team = NULL,
+
+    #' @field repos_dt An output table of repositories.
+    repos_dt = NULL,
+
+    #' @field commits_dt An output table of commits.
+    commits_dt = NULL,
+
+    #' @description Method to set connections to Git platforms
+    #' @param api_url A character, url address of API.
+    #' @param token A token.
+    #' @param orgs A character vector of organisations (owners of repositories
+    #'   in case of GitHub and groups of projects in case of GitLab).
+    #' @param set_org_limit An integer defining how many orgs API may pull.
+    #' @return Nothing, puts connection information into `$clients` slot
+    set_connection = function(api_url,
+                              token,
+                              orgs = NULL,
+                              set_org_limit = 300) {
+
+      if (grepl("github", api_url)) {
+        message("Set connection to GitHub.")
+        new_client <- GitHub$new(
+          rest_api_url = api_url,
+          token = token,
+          orgs = orgs,
+          org_limit = set_org_limit
+        )
+      } else if (grepl("https://", api_url) && grepl("gitlab|code", api_url)) {
```

Comment:
whe is the `https://` condition only used for gitlab?

Reply:
Good point, I overlooked it.